TTWWW-471: QA Updates

mattga11ant opened the pull request:
The first update was to make the prevalence min and max inputs to be a number input instead of a text input. This solves the issue of users being able to enter non-numerical values here.

Second update was to add functionality to unclick a map dot. This lead to me refactoring this functionality, as it was messy and had redundancy. Now there are separate functions for hover on, hover off, pin, and unpin.

These are the use cases I considered for the map and timeline dot clicked and hover states:
1. I hover a dot on the map, the tooltip for that dot on the map and on the timeline show. When I hover off, both hide.
2. I hover a dot on the timeline, the tooltip for that dot on the timeline and the map show. When I hover off, both hide.
3. I click a dot on the map. The tooltip on the map and the timeline show. If I click that dot again, both the map and timeline tooltips hide.
4. While there is an active clicked dot, so there is a tooltip showing on the map and timeline, then I hover over another dot on the map or the timeline. The active tooltip for the clicked dot on the map and the timeline do not change or hide. The tooltip for the hovered dot on the map and the timeline also show though, but hide when hovered off.
5. While there is an active clicked dot, so there is a tooltip showing on the map and timeline, then I click another dot on the map or the timeline. Both of the previous active tooltips hide, and the new active tooltips for the new clicked dot both show.

- [x] check out this branch
- [x] compile front end updates with npm run dev
- [x] try to enter a non-numeric value into the min and max prevalence fields, and confirm that you can't (should only be able to enter numbers)
- [x] click a map dot, confirm that it shows the tooltip on the map and the timeline, and changes the dot to the blue background with white border. 
- [x] click the active map dot, confirm that it hides the tooltip on the map and timeline, and reverts the map dot to the red background and border.
- [x] hover a map dot, confirm that it shows the tooltip and turns the dot black.
- [x] un-hover that same dot and confirm it hides the tooltip and the dot goes back to red background and red border.

yjfnyc reviewed (changes requested):
In terms of functionality, worked as described mostly except for firefox, text can still be input other than numbers.

I have some code comments, please see if they make sense to you.

Some issue I found that are not directly related to this ticket, please see if you can address here. 
1. **main.ts** has typo for function call `ttInitList`();
2. **list.ts** hasn't exported the ttInitList function.
3. the two file under folder **spectrum/autism_prevalence_map/static/svg** should be moved to folder **spectrum/autism_prevalence_map/static/autism_prevalence_map/img**, and related code adjusted, and folder **spectrum/autism_prevalence_map/static/svg** should be removed.
Hopefully after you address above-mentioned issues, we can close this 2 tickets. Thanks, @mattga11ant 

One follow-up about the svg files moving to the img folder, I did some research, The following setting added would work without changing using template tag **svg**, FYI
`SVG_DIRS = [(f"{BASE_DIR}" + "/autism_prevalence_map/static") + "/autism_prevalence_map/img/"]`

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/tailwind/tailwind.css:
```diff
@@ -392,4 +392,16 @@
+        -webkit-appearance: none;
+            margin: 0;
+        }
+        -moz-appearance: textfield;
```
I have addressed this in a different way now. Could not get Firefox to adhere to normal rules for a number field, and found this was a known issue with Firefox. Found a different suggested way to accomplish this that requires a little JavaScript.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -763,81 +736,82 @@ export function ttInitMap() {
-                .style('stroke-width', '0.5');
+                .style('stroke', '#910E1C');
+            d3.select('#timeline_dot_' + pk)
+                .style('fill', pointColor)
```
Good catch, I have this updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -763,81 +736,82 @@ export function ttInitMap() {
-            // show tooltip on timeline
-            showTimelineTooltip(d);
+            // hide the info card
+            hideCard();
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -884,20 +858,33 @@ export function ttInitMap() {
-            hideTimelineTooltip();
         }
 
+        function viewMoreInfo(d) {
```
This has been updated.

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/map.ts:
```diff
@@ -884,20 +858,33 @@ export function ttInitMap() {
+        }
+
+        // listener to close card
+        $('#card-close').click(function () {
```
This has been removed.

mattga11ant commented:
> In terms of functionality, worked as described mostly except for firefox, text can still be input other than numbers.
> 
> I have some code comments, please see if they make sense to you.
> 
> Some issue I found that are not directly related to this ticket, please see if you can address here.
> 
> 1. **main.ts** has typo for function call `ttInitList`();
> 2. **list.ts** hasn't exported the ttInitList function.
> 3. the two file under folder **spectrum/autism_prevalence_map/static/svg** should be moved to folder **spectrum/autism_prevalence_map/static/autism_prevalence_map/img**, and related code adjusted, and folder **spectrum/autism_prevalence_map/static/svg** should be removed.
>    Hopefully after you address above-mentioned issues, we can close this 2 tickets. Thanks, @mattga11ant
> 
> One follow-up about the svg files moving to the img folder, I did some research, The following setting added would work without changing using template tag **svg**, FYI `SVG_DIRS = [(f"{BASE_DIR}" + "/autism_prevalence_map/static") + "/autism_prevalence_map/img/"]`

1. I have the typo corrected
2. Added the export function
3. Moved these files and added the setting to point to the correct location
4. Related to number 3, when I moved the transmitter-logo.svg file, the fill color stopped working, so fixed that too

@yjfnyc all requested updates are complete.

yjfnyc reviewed (changes requested):
looks good to me. I have only code comment about one thing. Can you address it, then we can merge the code? Thanks!

mattga11ant commented on spectrum/autism_prevalence_map/static/autism_prevalence_map/ts/joint.ts:
```diff
@@ -401,5 +401,19 @@ export function ttInitJoint() {
         });
+
+        // make the min and max prevalence inputs accept numerical values only to account for Firefox allowing non-numeric values in a number input
+        $('input.preva-input').on('input', function() {
```
I have updated this.

mattga11ant commented on spectrum/autism_prevalence_map/templates/autism_prevalence_map/action_buttons.html:
```diff
@@ -13,7 +13,7 @@
                         <label class='absolute top-1.75 left-3 text-xs text-med-navy uppercase' for='prevalence-min'>Min</label>
-                        <input type='text' id='prevalence-min' class='w-10.5 pt-5.2 pb-1.25 bg-light-navy-2 rounded-sm-l text-2xs text-center text-med-navy'>
+                        <input type='text' inputmode='numeric' pattern='^[0-9]+$' title='Please enter whole numbers only' id='prevalence-min' class='w-10.5 pt-5.2 pb-1.25 bg-light-navy-2 rounded-sm-l text-2xs text-center text-med-navy preva-input'>
                     </div>
```
I have updated this.

mattga11ant commented on spectrum/autism_prevalence_map/templates/autism_prevalence_map/action_buttons.html:
```diff
@@ -22,7 +22,7 @@
                         <label class='absolute top-1.75 left-2.75 text-xs text-med-navy uppercase' for='prevalence-max'>Max</label>
-                        <input type='text' id='prevalence-max' class='w-10.5 pt-5.2 pb-1.25 bg-light-navy-2 rounded-sm-r text-2xs text-center text-med-navy'>
+                        <input type='text' inputmode='numeric' pattern='^[0-9]+$' title='Please enter whole numbers only' id='prevalence-max' class='w-10.5 pt-5.2 pb-1.25 bg-light-navy-2 rounded-sm-r text-2xs text-center text-med-navy preva-input'>
                     </div>
```
I have updated this.

yjfnyc reviewed (changes requested):
looks good. 

Can you limit the min and max number input within 0-500 inclusive, now they are unlimited.

mattga11ant commented:
> looks good.
> 
> Can you limit the min and max number input within 0-500 inclusive, now they are unlimited.

This is complete. Added to the JS function so that if  number less than 0 is input, it changes to 0, and if a value greater than 500 in input, it changes to 500.

yjfnyc reviewed (approved):
looks good to me now.
